src/models/json-api.model.spec.ts

```diff
@@ -149,4 +149,24 @@ describe('JsonApiModel', () => {
       });
     });
   });
+
+  describe('hasDirtyAttributes & rollbackAttributes', () => {
+    const author = new Author(datastore, {
+      id: '1',
+      attributes: {
+        name: 'Daniele'
+      }
+    });
+
+    it ('should return that has dirty attributes', () => {
+      author.name = 'New Name';
+      expect(author.hasDirtyAttributes).toBeTruthy();
+    });
+
+    it ('should to rollback to the initial author name', () => {
+      author.rollbackAttributes();
```

Comment:
Shouldn't the authors name be changed before you roll it back? Otherwise you do not test if the author name is really rolled back.

Reply:
No, we shouldn't because it's in the same scope and every change we do on the author object affects on every it() in this describe().